CardanoSharp.Wallet/AddressService.cs

```diff
@@ -36,11 +36,15 @@ public Address GetAddress(PublicKey payment, PublicKey stake, NetworkType networ
                     Buffer.BlockCopy(stakeEncoded, 0, addressArray, paymentEncoded.Length + 1, stakeEncoded.Length);
                     break;
                 case AddressType.Enterprise:
-                case AddressType.Reward:
                     addressArray = new byte[1 + paymentEncoded.Length];
                     addressArray[0] = header;
                     Buffer.BlockCopy(paymentEncoded, 0, addressArray, 1, paymentEncoded.Length);
                     break;
+                case AddressType.Reward:
+                    addressArray = new byte[1 + stakeEncoded.Length];
```

Comment:
Yup this does the trick. To address #43 we should allow our consumers to pass in a nullable payment public key and shift the encoding code down because :20 will throw a null reference exception. Same thing goes for stake public key when generating enterprise addresses. Also worth thinking about separate methods for address types for a cleaner API?